### PR TITLE
merge coreth 0.8.10-rc5 without fastsync & metrics package

### PR DESCRIPTION
**Pull request by ceyonur**
Merges coreth 0.8.10-rc5 without following fastsync & metrics commits:
1-https://github.com/ava-labs/coreth/commit/98e5171e5c9496a6c61c7c374a20c1e52352f31d
2-https://github.com/ava-labs/coreth/commit/0ec59e863df4a205217b40d0a6225078fd6bf515
3-https://github.com/ava-labs/coreth/commit/0aaaab6051efd2117dc4f09cb65c4e34a06dfc19
4-https://github.com/ava-labs/coreth/commit/0d4d96b7a7228f8dfd61b873cd499097261db363
5-https://github.com/ava-labs/coreth/commit/3bd3b56842354a7e3338b44504dd8a15904aa379
6-https://github.com/ava-labs/coreth/commit/2c40d11a8f75611c68449d60521022c0d7e986e7
7-https://github.com/ava-labs/coreth/commit/b2e08f00fa6d3af6982e88d448c691403ef480c8

### CODE REVIEW

File: core/genesis.go [hunk 1]
```diff
@@ -184,10 +184,12 @@ func SetupGenesisBlock(db ethdb.Database, genesis *Genesis) (*params.ChainConfig
 	if genesis.Config == nil {
 		return nil, errGenesisNoConfig
 	}
-	// Make sure genesis gas limit is consistent
-	gasLimitConfig := genesis.Config.FeeConfig.GasLimit.Uint64()
-	if gasLimitConfig != genesis.GasLimit {
-		return nil, fmt.Errorf("gas limit in fee config (%d) does not match gas limit in header (%d)", gasLimitConfig, genesis.GasLimit)
+	// Make sure genesis gas limit is consistent in SubnetEVM fork
+	if genesis.Config.IsSubnetEVM(new(big.Int).SetUint64(genesis.Timestamp)) {
```
Comment (patrick-ogrady) [inline core/genesis.go:188]: This should always happen. Not just after SubnetEVM fork IMO

**Review by aaronbuchwald — APPROVED** [review]
LGTM - great work!

File: params/config.go [hunk 2]
```diff
@@ -113,8 +113,8 @@ type ChainConfig struct {
 
 	SubnetEVMTimestamp *big.Int `json:"subnetEVMTimestamp,omitempty"` // A placeholder for the latest avalanche forks (nil = no fork, 0 = already activated)
 
-	FeeConfig          *FeeConfig `json:"feeConfig,omitempty"`
-	AllowFeeRecipients bool       `json:"allowFeeRecipients,omitempty"` // Allows fees to be collected by block builders.
+	FeeConfig          FeeConfig `json:"feeConfig,omitempty"`
```
Comment (patrick-ogrady) [inline params/config.go:116]: is `omitempty` valid here?

File: core/tx_pool.go [hunk 3]
```diff
@@ -663,15 +663,31 @@ func (pool *TxPool) local() map[common.Address]types.Transactions {
...
+	}
+
+	txNonce := tx.Nonce()
 	// Ensure the transaction adheres to nonce ordering
-	if currentNonce, txNonce := pool.currentState.GetNonce(from), txNonce; currentNonce > txNonce {
+	if currentNonce := pool.currentState.GetNonce(from); currentNonce > txNonce {
 		return fmt.Errorf("%w: address %s current nonce (%d) > tx nonce (%d)",
 			ErrNonceTooLow, from.Hex(), currentNonce, txNonce)
 	}
+	isTxAllowList := pool.chainconfig.IsTxAllowList(pool.currentHead.Number)
```
Comment (patrick-ogrady) [inline core/tx_pool.go:682]: Why do we need to create a var here? Shouldn't we just put this in the if statement.

File: params/config.go (hunk 2 above)
Comment (aaronbuchwald) [reply params/config.go:116]: No: https://www.sohamkamani.com/golang/omitempty/

File: core/genesis.go (hunk 1 above)
Comment (aaronbuchwald) [reply core/genesis.go:188]: It breaks a few tests, so going to leave as is for now.

Comment (aaronbuchwald) [reply core/genesis.go]: Put up a fix, but this check is required if we want to support pre-SubnetEVM upgrade functionality.

Comment (aaronbuchwald) [reply core/genesis.go]: Need to revert the last two commits if we want to maintain this.

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**